go/database/mpt/proof_test.go

```diff
@@ -0,0 +1,341 @@
+// Copyright (c) 2024 Fantom Foundation
+//
+// Use of this software is governed by the Business Source License included
+// in the LICENSE file and at fantom.foundation/bsl11.
+//
+// Change Date: 2028-4-16
+//
+// On the date above, in accordance with the Business Source License, use of
+// this software will be governed by the GNU Lesser General Public License v3.
+
+package mpt
+
+import (
+	"github.com/Fantom-foundation/Carmen/go/common"
+	"github.com/Fantom-foundation/Carmen/go/database/mpt/shared"
+	"go.uber.org/mock/gomock"
+	"testing"
+)
+
+func TestWitnessProof_Extract_and_Merge_Proofs(t *testing.T) {
+	ctrl := gomock.NewController(t)
+
+	address1 := common.Address{0xAB, 0xCD, 0xEF}
+	address2 := common.Address{0xAF, 0xCD, 0xEF}
+	key1 := common.Key{0x12, 0x34, 0x56, 0x78}
+	key2 := common.Key{0x23, 0x45, 0x67, 0x78}
```

Comment:
nit: since addresses and keys are hashed, simpler values like
```suggestion
	address1 := common.Address{1}
	address2 := common.Address{2}
	key1 := common.Key{1}
	key2 := common.Key{2}
```
would be sufficient. The extra size leaves the impression that those values are essential for the test.